integration-tests/auto-restore-mocks/with-auto-restore/__tests__/index.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const TestClass = require('../');
+const localClass = new TestClass();
+
+test('first test', () => {
+  jest.spyOn(localClass, 'test').mockImplementation(() => 'ABCD');
+  expect(localClass.test()).toEqual('ABCD');
+  expect(localClass.test.mock.calls.length).toBe(1);
```

Comment:
`.toHaveBeenCalledTimes(1)`

Reply:
Same with the others

Reply:
Ah, yes. I left that the way it was in the tests for resetMocks. I wasn't sure if it was using that older style for a specific reason or not. Fixed. 

As for the changelog, would this be correct? The changes touch jest-cli, jest-config, jest-jasmine2, etc. so I'm not sure which one to place in the beginning of the line. I went with jest-config since it's more related to the config.

```
* `[jest-config]` Add restoreMocks config option.
  ([#5327](https://github.com/facebook/jest/pull/5327))
```